python/cudf/cudf/core/reshape.py

```diff
@@ -198,16 +198,14 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
         elif isinstance(o, cudf.Series):
             typs.add(cudf.Series)
         else:
-            raise ValueError(f"cannot concatenate object of type {type(o)}")
+            raise TypeError(f"cannot concatenate object of type {type(o)}")
```

Comment:
Changed these exception types to `TypeError` as that is more appropriate and matches `pd.concat`